### PR TITLE
Context manager for selecting content DB 

### PR DESCRIPTION
**Pull request by jamalex**
## Summary

Adds a context manager/decorator to select content databases across a given set of database operations.
## TODO
- [ ] Have tests been written for the new code? _Some, but more coverage still needed_
- [x] Has documentation been written/updated? _Inline docs_
- [x] New dependencies (if any) added to requirements file

### CODE REVIEW

File: kolibri/content/content_db_router.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+"""
```
Comment (jamalex) [inline kolibri/content/content_db_router.py:1]: This is the file where most of the work is done.

**Comment by rtibbles** [comment]
@jamalex There were some weird conflicts here, as I think I had rebased my branch. Cherry picked it in, but unfortunately stole the blame for one of the commits in the merge conflict resolution :/

Is in my branch now.

**Comment by jamalex** [comment]
Oh, ok -- I did some rebasing against your `js_api` branch, but I guess you rebased again after that. :baseball: As long as it all makes it in!

Note on this stuff: it works for most cases. One problem (which I largely mitigated, but might still surface) is that Django's DB routing gets called at the time a queryset is _resolved_, not the time it's created, so if a queryset is created inside the context manager and then returned to elsewhere before lazy resolution occurs, it's possible it will throw an error. I think the cases in which this would happen should be rare now, with some changes I made and through upgrading to the bleeding edge version of django-mptt that includes this fix from a week ago: https://github.com/django-mptt/django-mptt/pull/478

**Comment by rtibbles** [comment]
Yeah - I have further mitigated this issue by removing any mention of specific databases within the content API methods - so now they have to be called from within a context manager.

The one case where this has still cropped up is in the Django REST framework Serializers, where a queryset is returned from methods and then further processing done. Eli has worked around this by continuing to use the `.using` method in this context.

**Pull request closed without merging**